### PR TITLE
Stabilize integration tests

### PR DESCRIPTION
**Pull request by daka1510**
### Scope
I checked the integration test results after https://github.com/Qiskit/qiskit-ibm-runtime/pull/114 was integrated.

#### Problems

1. in-progress matrix jobs are cancelled if any test fails
    - e.g. if integration tests against `cloud-production` fail, the integration tests against the other environments are cancelled
2. [Run integration test - legacy staging](https://github.com/Qiskit/qiskit-ibm-runtime/runs/5001253383?check_suite_focus=true) fails
```
Traceback (most recent call last):
  File "/home/runner/work/qiskit-ibm-runtime/qiskit-ibm-runtime/test/utils/decorators.py", line 100, in _wrapper
    raise Exception("Configuration Issue")
Exception: Configuration Issue
```
3. [Run integration test - cloud production](https://github.com/Qiskit/qiskit-ibm-runtime/runs/5001253430?check_suite_focus=true) fails
```
qiskit_ibm_runtime.api.exceptions.RequestsApiError: "HTTPSConnectionPool(host='us-east.quantum-computing.us-east.quantum-computing.cloud.ibm.com', port=443): Max retries exceeded with url: /devices (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f45f9c9be20>: Failed to establish a new connection: [Errno -2] Name or service not known'))"
```
4. Status Badge is broken
image

#### Related Solutions

- [X] 1./4. - This PR
  - @rathishcholarajan please review and merge 
- [x] 2. - Update environment configuration
  - @rathishcholarajan please add an environment configuration for `legacy-staging`
- [x] 3. - Update environment configuration
  - @rathishcholarajan please update the URL in the `cloud-production` environment as per https://github.com/Qiskit/qiskit-ibm-runtime/pull/118

### CODE REVIEW

**Comment by rathishcholarajan** [comment]
I've added `legacy-staging` and `cloud-staging` envs. Update URL in `cloud-production`.

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ jobs:
     name: Run integration tests - ${{ matrix.environment }}
     runs-on: ${{ matrix.os }}
     strategy:
+      # avoid cancellation of in-progress jobs if any matrix job fails
+      fail-fast: false
       matrix:
         python-version: [ 3.9 ]
         os: [ "ubuntu-latest" ]
```
Comment (rathishcholarajan) [inline .github/workflows/integration-tests.yml:28]: Could you please add `cloud-staging` as well in below line? (I am not able to comment on a line that has not changed)

**Review by rathishcholarajan — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (daka1510) [reply .github/workflows/integration-tests.yml:28]: Done via https://github.com/Qiskit/qiskit-ibm-runtime/pull/135/commits/44a9ae9904c7248e7918f1adea698829c5bc37d9

**Review by rathishcholarajan — COMMENTED** [review]
LGTM! Thanks.

**Pull request merged**